vaadin-date-picker-flow-parent/vaadin-date-picker-flow/src/main/java/com/vaadin/flow/component/datepicker/DatePicker.java

```diff
@@ -580,6 +580,21 @@ protected boolean isInputValuePresent() {
         return getElement().getProperty("_hasInputValue", false);
     }
 
+    @Override
+    public void setValue(LocalDate value) {
+        LocalDate oldValue = getValue();
+
+        super.setValue(value);
+
+        if (oldValue == null && value == null && isInputValuePresent()) {
```

Comment:
Should this compare against `getEmptyValue`?

Reply:
Good point. Let me update it.

Reply:
Done.